bin/webpack-dev-server.js

```diff
@@ -115,58 +112,7 @@ function startDevServer(config, options) {
     throw err;
   }
 
-  if (options.socket) {
-    server.listeningApp.on('error', (e) => {
-      if (e.code === 'EADDRINUSE') {
-        const clientSocket = new net.Socket();
-
-        clientSocket.on('error', (err) => {
-          if (err.code === 'ECONNREFUSED') {
-            // No other server listening on this socket so it can be safely removed
-            fs.unlinkSync(options.socket);
-
-            server.listen(options.socket, options.host, (error) => {
-              if (error) {
-                throw error;
-              }
-            });
-          }
-        });
-
-        clientSocket.connect({ path: options.socket }, () => {
-          throw new Error('This socket is already used');
-        });
-      }
-    });
```

Comment:
I think we can solve this without new method, it is very inconsistently with node api for servers, we can move this code without new method, just listen `error` event and handle this inside server class

Reply:
Do you mean handle it inside the `listen(port, hostname, fn)` method, rather than making a new method?